src/transport/multicast/link/rx.c

```diff
@@ -51,39 +51,43 @@ int8_t _z_multicast_recv_t_msg_na(_z_transport_multicast_t *ztm, _z_transport_me
     _z_mutex_lock(&ztm->_mutex_rx);
 #endif  // Z_MULTI_THREAD == 1
 
-    // Prepare the buffer
-    _z_zbuf_reset(&ztm->_zbuf);
+    do {
+        size_t to_read = 0;
```

Comment:
A minor comment. At protocol level Zenoh defines that a batch is maximum 2^16 -1 bytes. This hence could be a uint16_t.

Reply:
Created issue #193 to address this in a different PR.